mesonbuild/backend/backends.py

```diff
@@ -594,6 +596,24 @@ def get_executable_serialisation(
             exe_wrapper = None
 
         workdir = workdir or self.environment.get_build_dir()
+
+        # Must include separators as well
+        needs_rsp_file = can_use_rsp_file and sum(len(i) + 1 for i in cmd_args) >= get_rsp_threshold()
+
+        if needs_rsp_file:
+            hasher = hashlib.sha1()
+            if sys.version_info >= (3, 8):
+                args = shlex.join(cmd_args)
+            else:
+                args = ' '.join(shlex.quote(arg) for arg in cmd_args)
```

Comment:
This is incorrect, you need to use `mesonlib.quote_arg()` which is implemented in `mesonbuild/utils/universal.py`.

POSIX sh parsing/splitting is completely different from windows argument parsing/splitting.